account_payment_mode_auto_reconcile/models/account_payment_mode.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models, fields
+
+
+class AccountPaymentMode(models.Model):
+
+    _inherit = "account.payment.mode"
+
+    auto_reconcile_outstanding_credits = fields.Boolean(
+        string="Auto reconcile",
+        help="Reconcile automatically outstanding credits when an invoice "
+             "using this payment mode is validated, or when this payment mode "
+             "is defined on an open invoice."
+    )
+    auto_reconcile_allow_partial = fields.Boolean(
+        default=True,
+        string="Allow partial",
+        help="Allows automatic partial reconciliation of outstanding credits",
+    )
+    auto_reconcile_same_journal = fields.Boolean(
```

Comment:
instead of forcing reconciliation only under one journal it should fall to best match based on amount and date 
how for instance it will run reconciliation if under one journal there will be 2 payments for 200 and 300 with invoice on 300?
and forcing to one journal will exclude reconciliation of payment with different currencies, which can be useful, it's better to match partial payment in different currency than didn't match anything

Reply:
Really, this a request from the consultants that have been thinking hard to please their customer. I implemented the solution they thought was the best fit.
Also this is only an option that can be activated or not !